src/main/java/org/springframework/data/elasticsearch/core/AbstractElasticsearchTemplate.java

```diff
@@ -868,4 +909,38 @@ public ReadSearchScrollDocumentResponseCallback(Class<T> type, IndexCoordinates
 		}
 	}
 	// endregion
+
+	// region routing
+	@Override
+	public void setRoutingResolver(RoutingResolver routingResolver) {
+
+		Assert.notNull(routingResolver, "routingResolver must not be null");
+
+		this.routingResolver = routingResolver;
+	}
+
+	@Override
+	public RoutingResolver getRoutingResolver() {
+		return Objects.requireNonNull(routingResolver);
```

Comment:
`Objects.requireNonNull` throws a `NPE`. How can it happen that `RoutingResolver` is null?

Reply:
it cannot happen from the program logic. The property is set during instance initialization and afterwards it cannot be set to null. But as it is not initialized in a constructor, the property is `@Nullable`.

The whole initialization code will get cleaner (and this class removed) once we have Elasticsearch 8 and the transport client removed. Then we can make these fields not nullable and initialize them in the constructor.

For now I removed the `Objects.requireNonNull()` and added a comment to keep the compiler happy instead of using rNN